pkg/controller/nodes/task/handler.go

```diff
@@ -210,7 +216,14 @@ func (t *Handler) Setup(ctx context.Context, sCtx handler.SetupContext) error {
 		}
 		for _, tt := range p.RegisteredTaskTypes {
 			logger.Infof(ctx, "Plugin [%s] registered for TaskType [%s]", cp.GetID(), tt)
-			t.plugins[tt] = cp
+			t.defaultPlugins[tt] = cp
```

Comment:
I _think_ there is a boolean on `t` to the effect of `IsDefault` we should use that to determine which one is the default for that type

Reply:
oh interesting, the intended behavior then _isn't_ the current one where the last registered plugin for a task type is used? will this break anything if i change it in this PR?

Reply:
yeah :( it should have flat out thrown an exception (in the current behavior) but looks like that broke after propeller refactor we did last year...

Reply:
sorry I missed this reply! It looks like we're using the IsDefault as the global default task/plugin handler: https://github.com/lyft/flytepropeller/blob/master/pkg/controller/nodes/task/handler.go#L215 @EngHabu

Reply:
Yeah I guess using it for both purposes won't work either. The logic of using the last plugin as the default is very fragile. Depending on the ordering of the imports it can be one plugin or another. I do realize this is not introduced by your PR. Do you mind commenting on this line and filing a ticket to fix it?
We might need a way to configure these in propeller or something... P2 I guess :(

Reply:
sure, done here: https://github.com/lyft/flyte/issues/516